plugins/devlink/fu-devlink-component.c

```diff
@@ -60,21 +48,18 @@ fu_devlink_component_write_firmware(FuDevice *device,
 							  error);
 }
 
+static gboolean
+fu_devlink_component_probe(FuDevice *device, GError **error)
+{
+	/* optional */
+	fu_device_build_instance_id(device, NULL, "PCI", "VEN", "DEV", "COMPONENT", "PSID", NULL);
+	return fu_device_build_instance_id(device, error, "PCI", "VEN", "DEV", "COMPONENT", NULL);
```

Comment:
As I wrote above and in the previous patch comment, this looks wrong to me. 1) there is no PCI guarantee, could be backed by anything. 2) PSID is mlx5 specific, the aim is to support all drivers that expose various "fixed versions" to identify their device.